### PR TITLE
WEB - Password fields rework

### PR DESCRIPTION
**Pull request by H-chauvet**
This work is about password fields.
The goal is to add a password validator system everywhere passwords are manipulated but also to add a little icon to see the hidden password (Accessibility).

Pages concerned : Reset password, Change password after reset, Profile password change page, Login, Register 

Capture d’écran du 2024-10-06 16-09-24
Capture d’écran du 2024-10-06 16-09-03
Capture d’écran du 2024-10-06 16-08-54
Capture d’écran du 2024-10-06 16-07-41
Capture d’écran du 2024-11-07 20-54-00
Capture d’écran du 2024-11-07 20-54-53
Capture d’écran du 2024-11-08 11-42-53
Capture d’écran du 2024-11-08 11-42-42

### CODE REVIEW

File: Configurateur/front/lib/l10n/app_en.arb [hunk 1]
```diff
@@ -324,5 +325,12 @@
...
-}
+	"yourMessage": "Enter your message...",
+	"emailNotValid": "Enter a valid mail address.",
+	"atLeast": "At least - character",
+	"uppercaseLetters": "- Uppercase letter",
+	"numericCharacters": "- Numeric character",
+	"specialCharacters": "- Special character",
+	"lowercaseLetters": "- Lowercase letter",
+	"normalLetters": "- Classic letter"
+}
```
Comment (invpuppet) [inline Configurateur/front/lib/l10n/app_en.arb:336]: new line

File: Configurateur/front/lib/l10n/app_en.arb [hunk 2]
```diff
@@ -214,8 +215,8 @@
 	"objectPrice": "Object price",
 	"opinion": "Opinion",
 	"opinionConsultCreate": "Consult and give feedbaks on RISU!",
-	"opinionPage": "Opinion page",
-	"opinionPost": "Post an opinion",
+	"opinionPage": "Post a comment",
```
Comment (invpuppet) [inline Configurateur/front/lib/l10n/app_en.arb:218]: why create a duplicate

File: Configurateur/front/lib/l10n/app_fr.arb [hunk 3]
```diff
@@ -214,7 +215,7 @@
 	"objectPrice": "Prix de l'objet",
 	"opinion": "Avis",
 	"opinionConsultCreate": "Consulter et donner votre avis sur RISU !",
-	"opinionPage": "Page des avis",
+	"opinionPage": "Poster un avis",
```
Comment (invpuppet) [inline Configurateur/front/lib/l10n/app_fr.arb:218]: why create duplicate

File: Configurateur/front/lib/l10n/app_fr.arb [hunk 4]
```diff
@@ -323,5 +324,12 @@
...
-}
+	"yourMessage": "Entrez votre message...",
+	"emailNotValid": "Veuillez entrer un mail valide.",
+	"atLeast": "Au moins - caractères",
+	"uppercaseLetters": "- Lettres majuscules",
+	"numericCharacters": "- Chiffres",
+	"specialCharacters": "- Caractères spéciaux",
+	"lowercaseLetters": "- Lettres minuscules",
+	"normalLetters": "- Lettres classiques"
+}
```
Comment (invpuppet) [inline Configurateur/front/lib/l10n/app_fr.arb:335]: new line

File: Configurateur/front/lib/screens/company-profil/object-creation.dart [hunk 5]
```diff
@@ -25,6 +25,7 @@ import 'package:http/http.dart' as http;
 import 'package:http_parser/http_parser.dart';
 import 'package:image_picker/image_picker.dart';
 import 'package:provider/provider.dart';
+import 'package:front/components/custom_toast.dart';
```
Comment (invpuppet) [inline Configurateur/front/lib/screens/company-profil/object-creation.dart:28]: why add this when it's not used in the page

**Review by invpuppet — CHANGES_REQUESTED** [review]

File: Configurateur/front/lib/l10n/app_en.arb (hunk 2 above)
Comment (H-chauvet) [reply Configurateur/front/lib/l10n/app_en.arb:218]: there is no duplicates, just changing the value

File: Configurateur/front/lib/l10n/app_fr.arb (hunk 3 above)
Comment (H-chauvet) [reply Configurateur/front/lib/l10n/app_fr.arb:218]: No duplicates, value has changed

File: Configurateur/front/lib/screens/company-profil/object-creation.dart (hunk 5 above)
Comment (H-chauvet) [reply Configurateur/front/lib/screens/company-profil/object-creation.dart:28]: because it's used.

**Review by invpuppet — APPROVED** [review]

**Review by CedricCORGE — APPROVED** [review]

**Pull request merged**